src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/DiagnosticsTelemetryModule.cs

```diff
@@ -132,12 +183,104 @@ public void Initialize(TelemetryConfiguration configuration)
+
                         this.isInitialized = true;
                     }
                 }
             }
         }
-        
+
+        /// <summary>
+        /// Add a new Health Heartbeat property to the payload sent with each heartbeat.
+        /// 
+        /// To update the value of the property you are adding to the health heartbeat, 
+        /// <see cref="DiagnosticsTelemetryModule.SetHealthProperty"/>.
+        /// 
+        /// Note that you cannot add a HealthHeartbeatProperty with a name that already exists in the HealthHeartbeat
+        /// payload, including (but not limited to) the name of SDK-default items.
+        /// 
+        /// </summary>
+        /// <param name="propertyName">Name of the health heartbeat value to add</param>
+        /// <param name="propertyValue">Current value of the health heartbeat value to add</param>
+        /// <param name="isHealthy">Flag indicating whether or not the property represents a healthy value</param>
+        /// <returns>True if the new payload item is successfully added, false otherwise.</returns>
+        public bool AddHealthProperty(string propertyName, string propertyValue, bool isHealthy)
+        {
+            if (this.HeartbeatProvider != null)
+            {
+                try
+                {
+                    return this.HeartbeatProvider.AddHealthProperty(propertyName, propertyValue, isHealthy);
+                }
+                catch (Exception e)
+                {
+                    CoreEventSource.Log.LogError("Could not add heartbeat property. Exception: " + e.ToInvariantString());
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Set an updated value into an existing property of the health heartbeat.
+        /// 
+        /// After the new HealthHeartbeatProperty has been added (<see cref="DiagnosticsTelemetryModule.AddHealthProperty"/>) to the 
+        /// heartbeat payload, the value represented by that item can be updated using this method at any time.
+        /// 
+        /// </summary>
+        /// <param name="propertyName">Name of the health heartbeat payload item property to set the value and/or health status of.</param>
+        /// <param name="propertyValue">Value of the health heartbeat payload item. If this is null, the current value in the item is left unchanged.</param>
+        /// <param name="isHealthy">Health status of the health heartbeat payload item. If this is set to null the health status is left unchanged.</param>
+        /// <returns>True if the payload provider was added, false if it hasn't been added yet 
+        /// (<see cref="DiagnosticsTelemetryModule.AddHealthProperty"/>).</returns>
+        public bool SetHealthProperty(string propertyName, string propertyValue = null, bool? isHealthy = null)
+        {
+            if (this.HeartbeatProvider != null && !string.IsNullOrEmpty(propertyName) && (propertyValue != null || isHealthy != null))
```

Comment:
code would be much cleaner if you will pre-create the `this.HeartbeatProvider`. It not only will simplify code, but also will allow to set health properties before the `Initialize` called. Meaning you can set properties from other module's initialization code